test/test_nested_tensor_masking.py

```diff
@@ -39,312 +45,995 @@ def test_gen_nested_tensor(self):
+        check_tensor_options(self, tensor, a)
+
+        tensor, mask = a.to_tensor_mask(mask_dim=2)
+        self.assertEqual(tensor, torch.tensor([[1], [2], [3]]))
+        self.assertEqual(mask, torch.tensor([[True], [True], [True]]))
+        check_tensor_options(self, tensor, a)
+
+    def test_scalar_and_empty_nt(self):
+        a = nt.nested_tensor([
+            nt.nested_tensor([]),
+            nt.nested_tensor([
+                torch.tensor(11, dtype=torch.long)
             ])
         ])
 
-        # mask_dim = None
-        res_tensor, res_mask = original_nt.to_tensor_mask()
-        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
-                                   [[[[5, 6], [7, 8]]]]])
-        exp_mask = torch.tensor(True)
-        self.assertEqual(res_mask, exp_mask)
-        self.assertEqual(res_tensor, exp_tensor)
-
-        # mask_dim = 0
-        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=0)
-        self.assertEqual(res_mask, exp_mask)
-        self.assertEqual(res_tensor, exp_tensor)
-
-        # mask_dim = 1
-        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=1)
-        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
-                                   [[[[5, 6], [7, 8]]]]])
-        exp_mask = torch.tensor([True, True])
-        self.assertEqual(res_mask, exp_mask)
-        self.assertEqual(res_tensor, exp_tensor)
-
-        # mask_dim = 2
-        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=2)
-        exp_mask = torch.tensor([[True], [True]])
-        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
-                                   [[[[5, 6], [7, 8]]]]])
-        self.assertEqual(res_mask, exp_mask)
-        self.assertEqual(res_tensor, exp_tensor)
-        
-        # mask_dim = 3
-        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=3)
-        exp_mask = torch.tensor([[[True]], [[True]]])
-        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
-                                   [[[[5, 6], [7, 8]]]]])
-        self.assertEqual(res_mask, exp_mask)
-        self.assertEqual(res_tensor, exp_tensor)    
-
-        # mask_dim = 4
-        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=4)
-        exp_mask = torch.tensor([[[[True, True]]],[[[True, True]]]])
-        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
-                                   [[[[5, 6], [7, 8]]]]])
-        self.assertEqual(res_mask, exp_mask)
-        self.assertEqual(res_tensor, exp_tensor) 
+        tensor, mask = a.to_tensor_mask()
+        self.assertEqual(tensor, torch.tensor([[0], [11]], dtype=torch.long))
+        self.assertEqual(mask, torch.tensor([False,  True]))
+        check_tensor_options(self, tensor, a)
 
-    #
-    # Group of tests to test nested_tensor_from_tensor_mask() 
-    #
-    def test_nested_tensor_from_tensor_mask_dim_none(self):
-        original_nt = nt.nested_tensor([
+    def test_scalar_and_empty_nt_cuda(self):
```

Comment:
I'd codegen testcases that you want to repeat for different devices.

Reply:
Yeah, this is an issue. ideally, i need to do something similar to what we have in test_torch.py in pytorch. we should iterate through the dtypes and devices. will follow up as a separate PR to make this solution generic for all test scripts.
filed a task